src/general/StageHUDBase.cs

```diff
@@ -716,6 +736,52 @@ public void UpdateEnvironmentalBars(BiomeConditions biome)
         // TODO: pressure?
     }
 
+    /// <summary>
+    ///   Creates and displays a fossilisation button above each on-screen microbe.
+    /// </summary>
+    public void ShowFossilisationButtons()
+    {
+        var microbes = GetTree().GetNodesInGroup(Constants.AI_TAG_MICROBE).Cast<Microbe>();
+        foreach (var microbe in microbes)
+        {
+            if (microbe.Species is not MicrobeSpecies)
+                continue;
+
+            var button = FossilisationButtonScene.Instance<FossilisationButton>();
+            button.AttachedMicrobe = microbe;
+            button.Connect(nameof(FossilisationButton.OnFossilisationDialogOpened), this,
+                nameof(ShowFossilisationDialog));
+
+            // Display a faded button with a different hint if the species has been fossilised.
+            var alreadyFossilised = FossilisedSpecies.CreateListOfSaves().Contains(
+                microbe.Species.FormattedName + Constants.FOSSIL_EXTENSION_WITH_DOT);
+            button.AlreadyFossilised = alreadyFossilised;
+            button.HintTooltip = alreadyFossilised ?
+                TranslationServer.Translate("FOSSILISATION_HINT_ALREADY_FOSSILISED") :
+                TranslationServer.Translate("FOSSILISATION_HINT");
+
+            fossilisationButtonLayer.AddChild(button);
+        }
+    }
+
+    /// <summary>
+    ///   Destroys all fossilisation buttons on screen.
+    /// </summary>
+    public void HideFossilisationButtons()
+    {
+        fossilisationButtonLayer.QueueFreeChildren();
+    }
+
+    /// <summary>
+    ///   Opens the dialog to a fossilise the species selected with a given fossilisation button.
+    /// </summary>
+    /// <param name="button">The button attached to the microbe to fossilise</param>
+    public void ShowFossilisationDialog(FossilisationButton button)
+    {
+        fossilisationDialog.SelectedSpecies = button.AttachedMicrobe.Species;
```

Comment:
I think the fossilization dialog should use the base "Species" instead of referring to microbes as this is in the base HUD which means this needs to work for all future stages as well. I'm fine with if you specifically just put a "Not implemented for later stages" message in the fossilize dialog when opened for a non-microbe species.